core/src/main/java/org/elasticsearch/index/query/TermsQueryParser.java

```diff
@@ -140,14 +108,7 @@ public Query parse(QueryShardContext context) throws IOException, QueryParsingEx
                     throw new QueryParsingException(parseContext, "[terms] query lookup element requires specifying the path");
                 }
             } else if (token.isValue()) {
-                if (EXECUTION_KEY.equals(currentFieldName)) {
-                    // ignore
-                } else if (parseContext.parseFieldMatcher().match(currentFieldName, MIN_SHOULD_MATCH_FIELD)) {
-                    if (minShouldMatch != null) {
-                        throw new IllegalArgumentException("[" + currentFieldName + "] is not allowed in a filter context for the [" + TermsQueryBuilder.NAME + "] query");
-                    }
-                    minShouldMatch = parser.textOrNull();
-                } else if ("boost".equals(currentFieldName)) {
```

Comment:
Execution mode was actually ignored, while minShouldMatch was deprecated, and not available in the builder. Do we remove minShouldMatch as mentioned previous comments?